### PR TITLE
fix(api): validate responseOverride status code range in BackendTrafficPolicy

### LINKED ISSUE #8800: Fix int32/int64 API field handling
Values are being truncated such as in https://github.com/envoyproxy/gateway/issues/8797.

We should go through integer API fields and validate expected handling.

### PR DESCRIPTION
**Pull request by pujitha24**
**What this PR does / why we need it**:

`BackendTrafficPolicy`'s `responseOverride[].response.statusCode` field (`CustomResponse.StatusCode`,
`api/v1alpha1/shared_types.go`) is a `*int` with no kubebuilder validation. It is cast directly to
`uint32` during translation (`internal/gatewayapi/backendtrafficpolicy.go`:
`response.StatusCode = new(uint32(*ro.Response.StatusCode))`) with no bounds check. A negative value
such as `-1` wraps to `4294967295` after the cast.

Unlike the sibling `HTTPDirectResponseFilter.StatusCode` field, whose IR representation is validated
via `ir.CustomResponse.Validate()` (called from `HTTPRoute.Validate()`), the `ResponseOverride` IR
path has no `Validate()` call anywhere in `internal/ir` or `internal/gatewayapi`, so an invalid value
here is not caught at any later stage — it would flow straight into the generated Envoy config as a
`wrapperspb.UInt32Value`.

This is the same defect class already fixed for `GrpcStatus` (api: add cel validation of GrpcStatus)
and `MaxEjectionPercent` (api: add cel validation of MaxEjectionPercent): an integer API field lacking
the validation needed to safely narrow it to `uint32`.

**Approach**: Add `+kubebuilder:validation:Minimum=100` and `+kubebuilder:validation:Maximum=599` to
`CustomResponse.StatusCode`. This range exactly matches the bound already enforced at the IR layer in
`internal/ir/xds.go` (`CustomResponse.Validate()`: `*status > 599 || *status < 100`), so the fix moves
an existing constraint earlier, to admission time, rather than introducing a new one. Regenerated CRDs
and helm golden files via `make generate gen-check`.

**Validation**:
- `go build ./...` — passed.
- `make generate gen-check` — regenerated the CRD YAML/golden files included in this PR; no other
  drift.
- Added a new CEL-validation test case (`test/cel-validation/backendtrafficpolicy_test.go`,
  "response status code invalid in response override") asserting `statusCode: -1` is rejected.
  Verified it is a real regression test: with the API/CRD changes stashed, the test fails against a
  real envtest API server (`got err=nil; want error=[...should be greater than or equal to 100]`);
  with the changes restored, it passes. Ran via
  `make go.test.cel` — passed on all supported k8s versions (1.33.0, 1.34.1, 1.35.0, 1.36.0).
- `golangci-lint run` (repo config) on `./api/...` and `./internal/gatewayapi/...`, and on
  `test/cel-validation` (test submodule) — 0 issues.
- Not run locally: `tools/hack/*.sh` shellcheck step in `make lint` (pre-existing local environment
  issue: the pinned shellcheck release is x86_64-only and fails to exec under Rosetta-less arm64 —
  unrelated to this change, which touches no shell scripts).
- Most recent `Build and Test` run on `main` is green; `OSV-Scanner`/`Trivy` show pre-existing,
  unrelated failures on `main` (dependency/CVE scanning noise, not code).

**Which issue(s) this PR fixes**:
Fixes #

---

**PR Checklist**

- [x] **Authorship & ownership**: Coding agents / AI assistants are welcome, but I have reviewed every change, understand how and why it works, can explain and maintain it, and take full responsibility for this PR. I have not submitted generated output I do not understand.
- [x] **DCO**: All commits are signed off (`git commit -s`). See [DCO: Sign your work](https://gateway.envoyproxy.io/community/contributing/#dco-sign-your-work).
- [x] **API agreed first**: N/A — this is a minimal validation-tightening change to an existing field, not a new API surface, in the same style as previously merged PRs for the same issue.
- [x] **Required checks pass**: `make generate gen-check`, `make lint`, and the unit-test/coverage build pass. (Flaky e2e failures are not considered breakages, but `gen-check`, `lint`, and coverage **MUST** pass.) N/A note: the `shellcheck` sub-step of `make lint` could not be run locally due to a pre-existing local toolchain/arch mismatch unrelated to this change; `golangci-lint`, `kube-api-linter`, `yamllint`, and `codespell` all passed.
- [x] **Tests added/updated**: New/changed code is covered by appropriate tests.
- [x] **Docs**: N/A — this only tightens validation on an existing, already-documented field; no user-facing behavior/docs change beyond a clearer admission-time error for invalid input.
- [ ] **Release notes**: For any non-trivial change, added a release-note fragment under `release-notes/current/<section>/<pr-number>-<slug>.md` (see `release-notes/current/README.md` for sections and naming). N/A: the PR number needed for the filename isn't known until this PR is opened; happy to add it as a follow-up push once the number is assigned.
- [x] **Generated files committed**: Ran `make gen-check` and committed the result if API/helm charts/modules changed.
- [x] **Scope & compatibility**: The PR is reasonably scoped (no unrelated changes) and preserves backward compatibility, or any breaking change is called out above and documented in `release-notes/current/breaking_changes/`.
- [ ] **Codex review**: Requested a Codex review and addressed all of its comments.
- [ ] **Copilot review**: Requested a Copilot review and addressed all of its comments.

---
AI assistance: this change was drafted with Claude Code.

Fixes #8800

### CODE REVIEW

**Comment by pujitha24** [comment]
Just checking in on this one since it's been a few days — it's rebased and green, happy to make any changes if something would help move review along.

**Review by zirain — DISMISSED** [review]

File: api/v1alpha1/shared_types.go [hunk 1]
```diff
@@ -909,6 +909,8 @@ type CustomResponse struct {
 	// If unset, does not override the status of response.
 	//
 	// +optional
+	// +kubebuilder:validation:Minimum=100
```
Comment (arkodg) [inline api/v1alpha1/shared_types.go]: should be 200
https://github.com/envoyproxy/envoy/blob/8ea3b355532052e5e5a9a8fdfdb923f8075af302/api/envoy/extensions/http/custom_response/local_response_policy/v3/local_response_policy.proto#L38

**Comment by pujitha24** [comment]
Good catch, thanks. Fixed the minimum to 200 per the local_response_policy.proto bound, regenerated the CRD/helm goldens, updated the CEL test's expected error string, and added boundary cases for both a rejected 100 and an accepted 200.

**Review by arkodg — APPROVED** [review]

**Review by zirain — APPROVED** [review]

**Review by MengjiaLiang — APPROVED** [review]

**Comment by hsubramanianaks** [comment]
### Code review

LGTM

**Review by hsubramanianaks — APPROVED** [review]
